### PR TITLE
AVC: Added support for gaps in frame_num.

### PR DESCRIPTION
**Pull request by jeffqjiangNew**
- Fixed decode failure of conformance stream MR3_TANDBERG_B.264.

### CODE REVIEW

File: src/parser/avc_parser.cpp [hunk 1]
```diff
@@ -475,7 +485,7 @@ ParserResult AvcVideoParser::SendPicForDecode() {
         }
     }
 
-    for (i = buf_index; i < AVC_MAX_DPB_FRAMES; i++) {
+    for (i = buf_index; i < 16; i++) {
```
Comment (AryanSalmanpour) [inline src/parser/avc_parser.cpp:488]: why do we use `16` instead of the `AVC_MAX_DPB_FRAMES` here?

Comment (jeffqjiangNew) [reply src/parser/avc_parser.cpp:488]: AVC_MAX_DPB_FRAMES is increased from 16 to 18 to take care of the need to store more frames in DPB. But the ref_frames[16] in VAAPI picture parameter struct stays the same.

Comment (jeffqjiangNew) [reply src/parser/avc_parser.cpp:488]: We will reduce DPB buffer size back to 16 when we implement decode/display buffer pool.

File: src/parser/avc_defines.h [hunk 2]
```diff
@@ -33,7 +33,7 @@ THE SOFTWARE.
 
 #define AVC_MAX_REF_FRAME_NUM                           16
 #define AVC_MAX_REF_PICTURE_NUM                         32
-#define AVC_MAX_DPB_FRAMES                              16
+#define AVC_MAX_DPB_FRAMES                              18
```
Comment (rrawther) [inline src/parser/avc_defines.h:36]: why do we need to increase this to 18?

**Review by rrawther — CHANGES_REQUESTED** [review]
@jeffqjiangNew : Let's have a call to discuss this PR

File: src/parser/avc_defines.h (hunk 2 above)
Comment (jeffqjiangNew) [reply src/parser/avc_defines.h:36]: Due to current design limitations. I can explain in the call.

**Review by AryanSalmanpour — APPROVED** [review]

**Pull request merged**